sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/alibaba.yaml

```diff
@@ -64,30 +69,29 @@ provider:
         - text: |
             String aliyunAccessKeySecret = "100000000000000000000000000000";
           containsSecret: false
+
+    - rspecKey: S6336
+      id: alibaba-access-key-secrets-2
       detection:
+        pre:
+          include:
+            content:
+              - aliyun
+              - alibaba
+              - alicloud
         matching:
-          pattern: "(?i)(?<![A-Z])ali(?:yun|baba|cloud).{0,50}['\"`]([0-9a-z]{30})['\"`]"
+          pattern: "(?i)(?:SECRET_?(?:ACCESS)?_?KEY|(?:ACCESS)?_?KEY_?SECRET)\\b[^0-9a-z]{0,10}([0-9a-z]{30})(?![a-z0-9\\/+=$\\-_])"
         post:
           statisticalFilter:
             threshold: 4.2
           patternNot:
             - "EXAMPLE"
-
-    - rspecKey: S6336
-      id: alibaba-access-key-secrets-2
       metadata:
         name: Alibaba Access Key Secrets
         message: Make sure this Alibaba Cloud Access Key Secret gets revoked, changed, and removed from the code.
       examples:
         - text: |
+            # Alibaba key
             static string AccessKeySecret = "l0GdwcDYdJwB1VJ5pv0ormyTV9nhvW";
           containsSecret: true
           match: l0GdwcDYdJwB1VJ5pv0ormyTV9nhvW
-      detection:
-        matching:
-          pattern: "(?i)(?:SECRET_?(?:ACCESS)?_?KEY|(?:ACCESS)?_?KEY_?SECRET)\\b[^0-9a-z]{0,10}([0-9a-z]{30})(?![a-z0-9\\/+=$\\-_])"
-        post:
-          statisticalFilter:
-            threshold: 4.2
-          patternNot:
-            - "EXAMPLE"
```

Comment:
I think you mixed up the detection patterns between two different detection logics

Reply:
I am pretty sure that this is only a confusing diff display.
Basically we have one pattern with the ali.. prefix and one with the SECRET... prefix.
I added pre filters to both to decrease the false positives.